pennylane/transforms/mitigate.py

```diff
@@ -12,15 +12,116 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Provides transforms for mitigating quantum circuits."""
+from copy import copy
+
 from typing import Any, Dict, Optional, Sequence, Union
 
-from pennylane import QNode, apply
-from pennylane.math import mean, shape
+from pennylane import QNode, apply, adjoint
+from pennylane.math import mean, shape, round
 from pennylane.tape import QuantumTape
 from pennylane.transforms import batch_transform
 
+import pennylane as qml
+
+
+# @batch_transform
+def fold_global(circuit, scale_factor):
+    r"""Diffable global circuit folding function as is done in :`mitiq.zne.scaling.fold_global <https://mitiq.readthedocs.io/en/v.0.1a2/apidoc.html?highlight=global_folding#mitiq.zne.scaling.fold_global>`_
+
+    For a unitary ``circuit`` :math:`U = L_d .. L_1`, where :math:`L_i` can be either a gate or layer, ``fold_global`` implements
+
+    .. math:: \text{fold_global}(U) = U (U^\dagger U)^n (L^\adjoint_d L^\adjoint_{d-1} .. L^\adjoint_s) (L_s .. L_d)
+
+    where :math:`n = \lfloor (\lambda - 1)/2 \rfloor` and :math:`s = \lfloor \left((\lambda -1) \mod 2 \right) (d/2) \rfloor` are determined via the ``scale_factor`` :math:`=\lambda`.
+
+    TODO: add examples
+
+    """
+
+    if scale_factor < 1.0:
+        raise AttributeError("scale_factor must be >= 1")
+    assert scale_factor >= 1.0
+
+    # TODO: simplify this - currently just a workaround, problem is when tape contains adjoint(op)
+    def qfunc(op):
+        copy(op).queue()
+
+    # Generate base_circuit without measurements
+    # Treat all circuits as lists of operations, build new tape in the end
+    base_ops = circuit.expand().copy(copy_operations=True).operations
```

Comment:
When I try to make fold_global a `batch_transform` to act on qnodes, I get problems in this line. Any idea how to circumvent that problem of not being able to `expand()` a qnode? @josh146

Reply:
What exactly is the issue if `batch_transform` is used as a def? Could a `try-except` block be added here?

Reply:
I dont know what went wrong last time, but you are right, indeed there is no problem now in making it a batch_transform.